source/ProcessManager.Components/WorkingDays/DataHubCalendar.cs

```diff
@@ -0,0 +1,175 @@
+﻿// Copyright 2020 Energinet DataHub A/S
+//
+// Licensed under the Apache License, Version 2.0 (the "License2");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using NodaTime;
+
+namespace Energinet.DataHub.ProcessManager.Components.WorkingDays;
+
+public class DataHubCalendar
+{
+    private readonly IClock _clock;
+    private readonly DateTimeZone _zone;
+
+    public DataHubCalendar(IClock clock, DateTimeZone zone)
+    {
+        _clock = clock;
+        _zone = zone;
+    }
+
+    public Instant CurrentDate()
+    {
+        return _clock.GetCurrentInstant().InZone(_zone).ToInstant();
+    }
+
+    /// <summary>
+    /// Calculates the DataHub working date back in time relative to today.
+    /// </summary>
+    /// <param name="count">The number of days back in time. This figure must be between 0 and 100 inclusive.</param>
+    /// <returns>A DataHub working date relative to today.</returns>
+    public Instant GetWorkingDayRelativeToTodayBackInTime(int count)
```

Comment:
Maybe this would be more flexible if it just take a period (Maybe `Internval`) and then calculated the amount working days between those two dates?

Reply:
Wait i misunderstood this, i thought it counted the amount of working days. I'm unsure what it actually does, based on the description. Does it get the earliest "work day" that is at most `count` days ago?

Reply:
Maybe something like

```suggestion
   /// <summary>
   /// Subtracts X work days from the given date, and returns the resulting date (calculate the date that is X workdays ago from the given date)
   /// </summary>
    public Instant SubtractWorkdaysFromDate(Instant date, int subtractCount)
```